### PR TITLE
Preprocessing followup

### PR DESCRIPTION
**Pull request by Runi-VN**
Followup on remainders from #8

- [x] do a regex check [A-Za-z0-9] on hashtags/mentions before adding them to their dict.
- ~~some testing~~
- ~~error handling~~
- [x] Consider moving aggressive `A-Za-z`-regex further down the chain? After lemmatisation?
- [ ] rename get_tweet_data (wait until merge)
- [x] make a sort function for stats
- [x] update documentation, mentions still have 'hashtags' in it
- [x] remove unnecesary tweet_text check

Unable to fix weird cases like this atm:

image

### CODE REVIEW

File: modules/Preprocessing.py [hunk 1]
```diff
@@ -89,17 +107,21 @@ def get_tweet_data(tweets: List[Dict[str, str]]):
...
+                    # clean mention
+                    clean_word = re.sub(_REGEX_CHAR_MATCHER_TWEETS, "", word)
                     # add to local hashtags
-                    tweet['mentions'].append(word)
+                    tweet['mentions'].append(clean_word)
                     # add to overall hashtags
-                    mention_stats[word.lower()] = mention_stats.get(word.lower(), 0) + 1
+                    mention_stats[clean_word.lower()] = mention_stats.get(clean_word.lower(), 0) + 1
                     # remove mention
                     tweet_text = tweet_text.replace(word, '')
```
Comment (Runi-VN) [inline modules/Preprocessing.py]: Would love suggestions on how to handle this repeat code. I know there's an answer to it, maybe even without an external function, but it eludes me right now.

I've considered list comprehensions as well, but it does not allow handling data in the same way as we do now.

**Comment by Runi-VN** [comment]
Having some issues optimizing the above code. It's not really clean to move the `stats`-collections in and out of functions, or the text element either.

Hmm.

**Comment by MrHardcode** [comment]
Is this branch close to ready or ready to be merged with master?

**Comment by Runi-VN** [comment]
> Is this branch close to ready or ready to be merged with master?

Forgot to push, let me clean it up and it'll be ready.

File: modules/Preprocessing.py (hunk 1 above)
Comment (Runi-VN) [reply modules/Preprocessing.py]: No solution.

**Review by MrHardcode — APPROVED** [review]

**Review by MalteMagnussen — APPROVED** [review]

**Pull request merged**